src/compiler/types.rs

```diff
@@ -2265,8 +2253,35 @@ impl StringLiteralType {
     }
 }
 
+impl LiteralTypeInterface for StringLiteralType {
```

Comment:
This was a mistake in the previous PR, this isn't actually a "boilerplate" trait implementation because `.get_or_initialize_fresh_type()` has a custom implementation